### PR TITLE
Gp 3585.api shesmu plugin priority

### PR DESCRIPTION
**Pull request by grapigeau**
Jira ticket:

- [x] Includes a change file
- [ ] Updates developer documentation

### CODE REVIEW

**Review by callunity — APPROVED** [review]

File: changes/add_priority_target.md [hunk 1]
```diff
@@ -0,0 +1 @@
+priority by default to the consumableResource shesmu targets
```
Comment (callunity) [inline changes/add_priority_target.md:1]: What do you think about this wording for clarity? 
`priority as a consumable resource on al Vidarr targets`

(as theoretically more things than just Shesmu could make use of the targets)

File: vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java [hunk 2]
```diff
@@ -581,14 +581,17 @@ public void emit(SectionRenderer sectionRenderer)
...
                                           .map(
                                               name ->
                                                   new Pair<>(name, consumableResources.get(name))),
                                       Stream.of(
                                           new Pair<String, ConsumableResource>(
-                                              "", maxInFlightPerWorkflow)))
+                                              "", maxInFlightPerWorkflow))),
+                                    Stream.of(
+                                            new Pair<String, ConsumableResource>(
+                                                "priority", priorityPerWorkflow)))
```
Comment (avarsava) [inline vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java:594]: how's it work that maxInFlight is "" but priority has a name?

File: vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java [hunk 3]
```diff
@@ -581,14 +581,17 @@ public void emit(SectionRenderer sectionRenderer)
...
+                              Stream.concat(Stream.concat(
                                       e.getValue().getConsumableResources().stream()
                                           .map(
                                               name ->
                                                   new Pair<>(name, consumableResources.get(name))),
                                       Stream.of(
                                           new Pair<String, ConsumableResource>(
-                                              "", maxInFlightPerWorkflow)))
+                                              "", maxInFlightPerWorkflow))),
+                                    Stream.of(
```
Comment (avarsava) [inline vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java:592]: Stream.of takes variable values (https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Stream.html#of(T...)), have 1 Stream.concat and expand the existing Stream.of to include both new Pairs

File: vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java (hunk 2 above)
Comment (grapigeau) [reply vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java]: I'm not sure why it's "" either, other than it's simply never exposed so it doesn't matter?

**Review by avarsava — APPROVED** [review]

**Pull request merged**